### DISCUSSION TITLE
refresh query variable on datasource change

### DISCUSSION THREAD

**Discussion by t80027t** [grafana/grafana#54921] [Dashboards, Panels & Visualizations]
> In the Refresh list, select when the variable should update options.
On Dashboard Load - Queries the data source every time the dashboard loads. This slows down dashboard loading, because the variable query needs to be completed before dashboard can be initialized.
On Time Range Change - Queries the data source when the dashboard time range changes. Only use this option if your variable options query contains a time range filter or is dependent on the dashboard time range.

Given a dashboard variable can be added to change datasource, it would also be useful to add an query variable refresh option 'On Datasource Change', and then add the ability to combine e.g. On Time Range Change or On Datasource Change.

Alternatively the option would be part of the datasource variable properties itself, but would be restricted to refreshing all query variables on the same dashboard.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
